### DISCUSSION TITLE
Two updates for the same dependency mentioned in PR

### DISCUSSION THREAD

**Discussion by nyg** [renovatebot/renovate#14316] [Request Help]
Hello,

As you can see in this PR https://github.com/nyg/dependabot-vuln-viewer/pull/12, it mentions two updates for Next.js, one from 12.0.10 to 12.1.0 and one from 12.0.8 to 12.0.9. I don't think the second should be mentioned at all because the Next.js version was (when the PR was created) 12.0.10.

Is this a bug? (note that the commit correctly updates the Next.js version, no problem there)

nyg

**Comment by rarkins** [comment]
Would need to see logs before being able to comment

**Reply by nyg** [reply]
Yes, of course. I see that there are two Github vulnerability alerts for Next.js but, unless I'm mistaken, the first one shouldn't be here because the next version was 12.0.10 and thus doesn't fit in the `>= 12.0.0, < 12.0.9` range.

<details>
<summary>Relevant logs?</summary>

```
DEBUG: GitHub vulnerability details
{
  "alerts": {
    "npm/next": {
      ">= 12.0.0, < 12.0.9": "12.0.9",
      ">= 10.0.0, < 12.1.0": "12.1.0"
    },
    "npm/node-fetch": {
      "< 2.6.7": "2.6.7"
    }
  }
}
DEBUG: alert package rules
{
  "alertPackageRules": [
    {
      "matchDatasources": [
        "npm"
      ],
      "matchPackageNames": [
        "next"
      ],
      "matchCurrentVersion": "= 12.0.8",
      "matchFiles": [
        "package-lock.json"
      ]
    },
    {
      "matchDatasources": [
        "npm"
      ],
      "matchPackageNames": [
        "next"
      ],
      "matchCurrentVersion": "= 12.0.10",
      "matchFiles": [
        "package-lock.json"
      ]
    },
...
DEBUG: npm.updateLockedDependency: next@12.0.10 -> 12.1.0 [package-lock.json](branch="renovate-npm-next-vulnerability")
DEBUG: Found matching dependencies with length 1(branch="renovate-npm-next-vulnerability")
DEBUG: npm.updateDependency(): dependencies.next = 12.1.0(branch="renovate-npm-next-vulnerability")
DEBUG: npm.updateLockedDependency: next@12.0.8 -> 12.0.9 [package-lock.json](branch="renovate-npm-next-vulnerability")
```
</details>

**Reply by rarkins** [reply]
Double vulnerability alert with competing versions is definitely an edge case. If you would like to submit a PR which fixes it then go ahead, but I don't think it's worth tracking with an issue otherwise

**Reply by nyg** [reply]
Hello again,

I think I found the reason for this. TLDR: when you retrieve vulnerability alerts from GH, do you check their state (i.e. OPEN, FIXED, DISMISSED)?

If I go into the dashboard issue for the repository I mentioned, I can still see this:

> **Other Branches**
>
> [_] Update dependency next [SECURITY]
>
> **Ignored or Blocked**
> 
> [_]Update dependency node-fetch to 2.6.7 [SECURITY]

However, if I go to the _Security > Dependabot alerts_ tab of the repo, there are no open alerts, but three of them are closed. Here is the result of a GraphQL query I wrote to retrieve alerts:

<details>
<summary>GraphQL query</summary>

```graphql
query {
  repository(name: "dependabot-vuln-viewer", owner: "nyg") {
    vulnerabilityAlerts(first: 10) {
      totalCount
      nodes {
        securityVulnerability {
          package {
            name
          }
        }
        state
      }
    }
  }
}
```
</details>

<details>
<summary>Result</summary>

```json
{
  "data": {
    "repository": {
      "vulnerabilityAlerts": {
        "totalCount": 3,
        "nodes": [
          {
            "securityVulnerability": {
              "package": {
                "name": "next"
              }
            },
            "state": "FIXED"
          },
          {
            "securityVulnerability": {
              "package": {
                "name": "node-fetch"
              }
            },
            "state": "FIXED"
          },
          {
            "securityVulnerability": {
              "package": {
                "name": "next"
              }
            },
            "state": "FIXED"
          }
        ]
      }
    }
  }
}
```
</details>

I think renovate should only take into account OPEN alerts, no?

EDIT: I looked a bit at the code and think it's just a matter of adding a filter in [lib/platform/github/graphql.ts](https://github.com/renovatebot/renovate/blob/b130bf5afb054704b646d894e58eac86df760220/lib/platform/github/graphql.ts#L130):

```js
export const vulnerabilityAlertsQuery = `
query($owner: String!, $name: String!) {
  repository(owner: $owner, name: $name) {
    vulnerabilityAlerts(last: 100, states: OPEN) {
```

**Reply by rarkins** [reply]
I thought we did filtering of them later on after we get the results, but even if so then it would be a good idea to filter them with the query as you suggest. Feel free to submit a PR or feature request issue

**Reply by nyg** [reply]
Ok, I will try to provide a PR for this :). EDIT: #14431.
